File: src/net/dinkla/raytracer/objects/acceleration/kdtree/Leaf.java
package net.dinkla.raytracer.objects.acceleration.kdtree;

import net.dinkla.raytracer.hits.Hit;
import net.dinkla.raytracer.math.BBox;
import net.dinkla.raytracer.math.Ray;
import net.dinkla.raytracer.objects.GeometricObject;
import net.dinkla.raytracer.objects.acceleration.Grid;
import net.dinkla.raytracer.objects.compound.Compound;

import java.util.List;

/**
* Created by IntelliJ IDEA.
* User: jorndinkla
* Date: 11.06.2010
* Time: 09:40:41
* To change this template use File | Settings | File Templates.
*/
public class Leaf extends AbstractNode {
    protected final Compound compound;

    public Leaf(List<GeometricObject> objects) {
//        if (objects.size() > 1000) {
//            compound = new Grid();
//        } else {
            compound = new Compound();
//        }
        compound.add(objects);
        compound.initialize();
    }

    @Override
    public boolean hit(Ray ray, Hit sr) {
        return compound.hit(ray, sr);
    }

    @Override
    public BBox getBoundingBox() {
        return compound.getBoundingBox();
    }

    @Override
    public int size() {
        return compound.size();
    }

    @Override
    public String toString() {
        return "Leaf " + size() + " " + getBoundingBox(); 
    }

    @Override
    public String printBBoxes(int incr) {
        StringBuilder sb = new StringBuilder();
        for (int i=0; i<incr;i++) {
            sb.append(" ");
        }
        sb.append("-");
        return sb.toString();
    }
}

File: src/main/kotlin/net/dinkla/raytracer/objects/acceleration/kdtree/Leaf.kt
package net.dinkla.raytracer.objects.acceleration.kdtree

import net.dinkla.raytracer.hits.Hit
import net.dinkla.raytracer.math.BBox
import net.dinkla.raytracer.math.Ray
import net.dinkla.raytracer.objects.GeometricObject
import net.dinkla.raytracer.objects.acceleration.Grid
import net.dinkla.raytracer.objects.compound.Compound

class Leaf(objects: List<GeometricObject>) : AbstractNode() {
    protected val compound: Compound

    override val boundingBox: BBox
        get() = compound.boundingBox

    init {
        //        if (objects.size() > 1000) {
        //            compound = new Grid();
        //        } else {
        compound = Compound()
        //        }
        compound.add(objects)
        compound.initialize()
    }

    override fun hit(ray: Ray, sr: Hit): Boolean = compound.hit(ray, sr)

    override fun size(): Int = compound.size()

    override fun printBBoxes(incr: Int): String {
        val sb = StringBuilder()
        for (i in 0 until incr) {
            sb.append(" ")
        }
        sb.append("-")
        return sb.toString()
    }

    override fun toString(): String = "Leaf(${size()}, ${boundingBox}"
}
